### PR TITLE
Moving Archivist types into mage.archivist

### PR DESCRIPTION
**Pull request by stelcheck**
Additionally, also making the expirationTime on
certain calls optional.

### CODE REVIEW

**Comment by stelcheck** [comment]
@ronkorving question: are mediaType and encoding optional parameters?

File: types.d.ts [hunk 1]
```diff
@@ -1042,6 +1017,31 @@ declare var mage: Mage;
...
+        type ArchivistEncoding = 'utf8' | 'buffer' | 'live';
+
+        /**
+         * Callback functions
+         */
+        type ArchivistExistsCallback = (error: Error|null, doesValueExist: boolean) => void;
+
+        type ArchivistGetCallback<T> = (error: Error|null, value: T) => void;
+
+        type ArchivistMGetCallback<T> = (error: Error|null, value: T[]) => void;
```
Comment (ronkorving) [inline types.d.ts]: This can also be a key/value object, not just an array. Depends on the input to mget basically.

Comment (stelcheck) [inline types.d.ts]: Can you give me code samples?

Comment (ronkorving) [inline types.d.ts]:
```js
const queries = {
  user: { topic: 'user', index: [userId] },
  friend: { topic: 'user', index: [friendId] }
};

archivist.mget(queries, (error, values) => {
  // values.user
  // values.friend
});
```

Comment (stelcheck) [inline types.d.ts]: That then means we need to also create two types of queries - we only have one in the type def.

Are there official names for those two different types of queries?

Comment (ronkorving) [inline types.d.ts]: I don't believe there are.

**Pull request merged**